scipy/stats/tests/test_stats.py

```diff
@@ -581,6 +581,19 @@ def test_nan_policy(self):
         assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='raise')
         assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='foobar')
 
+    def test_nan_policy_bug_12458(self):
+        np.random.seed(5)
+        x = np.random.rand(5, 10)
+        k = 6
+        x[:, k] = np.nan
+        y = np.delete(x, k, axis=1)
+        corx, px = stats.spearmanr(x, nan_policy='omit')
+        cory, py = stats.spearmanr(y)
+        corx = np.delete(np.delete(corx, k, axis=1), k, axis=0)
+        px = np.delete(np.delete(px, k, axis=1), k, axis=0)
+        assert_allclose(corx, cory)
+        assert_allclose(px, py, atol=1e-14)
```

Comment:
It looks like these 2 lines are comparing differently, `px, py` are compared absolutely whereas `corx, cory` are compared relatively. Is there a reason to not make both comparisons on absolute terms?

Reply:
I don't really remember, but let's try it.